src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/court/VideoLinkBookingService.kt

```diff
@@ -304,19 +304,23 @@ class VideoLinkBookingService(
   }
 
   @Transactional
-  fun deleteAppointments(appointmentId: Long) {
+  fun processNomisUpdate(appointmentId: Long, recordDeleted: Boolean) {
     val videoLinkAppointment = videoLinkAppointmentRepository.findOneByAppointmentId(appointmentId)
-
-    if (videoLinkAppointment.isPresent && videoLinkAppointment.get().hearingType == MAIN) {
-      videoLinkBookingRepository.delete(videoLinkAppointment.get().videoLinkBooking)
-      val appointmentsToDelete = videoLinkAppointment.get().videoLinkBooking.appointments
-        .filter { it.value.appointmentId != appointmentId }
-        .map { it.value.appointmentId }
-      if (appointmentsToDelete.isNotEmpty()) {
-        prisonApiService.deleteAppointments(appointmentsToDelete, EventPropagation.DENY)
+    if (!videoLinkAppointment.isPresent) return
+    if (recordDeleted) {
+      if (videoLinkAppointment.get().hearingType != MAIN) {
+        videoLinkAppointmentRepository.delete(videoLinkAppointment.get())
+      } else {
+        videoLinkBookingRepository.delete(videoLinkAppointment.get().videoLinkBooking)
+        val appointmentsToDelete = videoLinkAppointment.get().videoLinkBooking.appointments
+          .filter { it.value.appointmentId != appointmentId }
+          .map { it.value.appointmentId }
+        if (appointmentsToDelete.isNotEmpty()) {
+          prisonApiService.deleteAppointments(appointmentsToDelete, EventPropagation.DENY)
+        }
       }
-    } else if (videoLinkAppointment.isPresent) {
-      videoLinkAppointmentRepository.delete(videoLinkAppointment.get())
+    } else {
+      videoLinkBookingEventListener.bookingUpdatedInNomis(videoLinkAppointment.get().videoLinkBooking)
```

Comment:
actually it would be best to pass the appointment through and the `appointmentChangedEventMessage`

we could do with having useful info appear in the properties map about both the updated appointment:
`update_bookingId`
`update_startDate`
`update_endDate`
`update_location`
`comment`

and also from the actual video link appointment:
`current_bookingId`
`current_startDate`
`current_endDate`
`current_location`
`current_comment`

This will require updating the props - I think it needs to be a map that is one level deep.